classify-split-extract-workflow/classify-job/split_and_classify.py

```diff
@@ -0,0 +1,358 @@
+    """Streams classification results to a specified callback URL."""
+    logger.info(f"bucket={bucket_name}, blob_object={file_name}")
+    success = bool(bucket_name and file_name)
+    result = "Classification Job completed successfully, proceed with extraction" if success else \
+        "Classification Job failed"
+
+    payload = {
+        "result": result,
+        "success": success,
+        "bucket": bucket_name,
+        "object": file_name,
+    }
+
+    utils.send_callback_request(call_back_url, payload)
+
+
+def save_classification_results(classified_items: Dict) -> Tuple[Optional[str], Optional[str]]:
+    """Saves classification results to Google Cloud Storage."""
+    payload_data = []
+    try:
+
+        for document_type, f_uris in classified_items.items():
+            model_name = config.get_model_name(document_type)
+            processor_name = config.get_parser_name_by_doc_type(document_type)
+            out_table_name = config.get_out_table_name(document_type)
+
+            processor, _ = docai_helper.get_docai_input(processor_name)
+            object_table_name = bq_mlops.object_table_create(f_uris=f_uris, document_type=document_type)
+            bq_mlops.remote_model_create(processor=processor, model_name=model_name)
+
+            payload_data.append({
+                "object_table_name": object_table_name,
+                "model_name": model_name,
+                "out_table_name": out_table_name
+            })
+
+        if not payload_data:
+            logger.warning("Payload data is empty, skipping")
+            return None, None
+
+        prefix = utils.get_utc_timestamp()
+
+        bucket, blob_object = gcs_helper.write_data_to_gcs(
+            bucket_name=config.CLASSIFY_OUTPUT_BUCKET,
+            blob_name=f"{prefix}_{config.OUTPUT_FILE_JSON}",
+            content=json.dumps(payload_data, indent=4),
+            mime_type='application/json'
+        )
+    except Exception as e:
+        logger.error(f"Exception while saving classification results, {e}")
+        return None, None
+
+    logger.info(f"Saved classification results to,"
+                f" bucket={bucket}, file_name={blob_object}")
+    return bucket, blob_object
+
+
+def is_splitting_required(entities):
+    try:
+        return not all(len(entity.page_anchor.page_refs) == 0 or
+                       all(not ref for ref in entity.page_anchor.page_refs) for entity in entities)
+    except AttributeError:
+        return False
+
+
+def check_confidence_threshold_passed(predicted_confidence: float) -> bool:
+    confidence_threshold = config.get_classification_confidence_threshold()
+    if predicted_confidence < confidence_threshold:
+        logger.warning(f"Confidence threshold not passed for "
+                       f"{predicted_confidence} < {confidence_threshold}")
+        return False
+    return True
+
+
+def split_pdf(gcs_uri: str, entities: List[Document.Entity]) -> Dict:
```

Comment:
Is this needed since the Toolbox error was resolved?

Reply:
That method is doing more than toolbox (e.g. updating the metadata of the GCS file). 
With toolbox a lot of info will be lost after splitting, so I cannot rely on it.